app/src/main/java/ch/epfl/polybazaar/FillListingActivity.java

```diff
@@ -275,4 +300,44 @@ private  void takePicture(){
     }
 
 
+    private boolean fillFieldsIfEdit() {
+        Bundle bundle = getIntent().getExtras();
+        if(bundle == null){
+            return false;
+        }
+        String listingID = bundle.getString("listingID", "-1");
+        if (listingID.equals("-1")){
+            return false;
+        }
+        Listing listing = (Listing)bundle.get("listing");
+
+        titleSelector.setText(listing.getTitle());
+        descriptionSelector.setText(listing.getDescription());
+        freeSwitch.setChecked(listing.getPrice().equals("0.0"));
+        priceSelector.setText(listing.getPrice());
+        Category editedCategory = new StringCategory(listing.getCategory());
+        traversingCategory = CategoryRepository.getCategoryContaining(editedCategory);
+        categorySelector.setSelection(CategoryRepository.indexOf(traversingCategory)+1);
+        return true;
+    }
+
+    private void deleteOldListingAndSubmitNewOne() {
+        if (!checkFields()) {
+            Toast.makeText(getApplicationContext(), INCORRECT_FIELDS_TEXT, Toast.LENGTH_SHORT).show();
```

Comment:
You might want to replace string constants like INCORRECT_FIELDS_TEXT by string values in res/values/strings.xml, it seems like this is the more conventional android approach. These values can also be referenced directly within layout files (probably not that useful in this situation though).